Lib/crypt.py

```diff
@@ -30,7 +30,13 @@ def mksalt(method=None, *, log_rounds=12):
     if not method.ident:
         s = ''
     elif method.ident[0] == '2':
+        if rounds is None:
+            log_rounds = 12
+        else:
+            log_rounds = (rounds-1).bit_length()
```

Comment:
According to the discussion in bpo-31702, ValueError should be raised in case `rounds` isn't a power of two.

Reply:
I agree. You must raise a VAlueError if it's not exactly a power of 2.

```
if (1 << log_rounds) != rounds: raise ValueError("rounds must be a power of 2, got %s" % rounds)
```